chainbench/test_data/solana.py

```diff
@@ -0,0 +1,101 @@
+from tenacity import retry, stop_after_attempt
+
+from chainbench.test_data.base import BaseTestData, Block, BlockchainData
+from chainbench.util.rng import get_rng
+
+
+class SolanaTestData(BaseTestData):
+    TXS_REQUIRED = 100
+    ACCOUNTS_REQUIRED = 200
+    BLOCK_TIME = 0.4
+
+    def _fetch_block(
+        self, block_number: int, return_txs: bool = True
+    ) -> dict | Exception:
+        if return_txs:
+            transaction_details = "accounts"
+        else:
+            transaction_details = "none"
+        config_object = {
+            "encoding": "json",
+            "transactionDetails": transaction_details,
+            "rewards": False,
+            "maxSupportedTransactionVersion": 0,
+        }
+        try:
+            result = self._make_call("getBlock", [block_number, config_object])
+        except Exception as e:
+            self._logger.error(f"Failed to fetch block {block_number}: {e}")
+            return e
```

Comment:
Why returning Exception instead of raising it here? As I see, it is raised it right after getting the result in one usage and just passed as normal value in the other.

Reply:
Because I want the retry to occur at one level above this, to obtain a new number when retry occurs. When I don't return exception here and raise it here instead, the retry at `_fetch_random_block` doesn't trigger.

Reply:
That's weird actually, because the exception raised in _fetch_block should pop in every function below it in stack, i.e. it should raise in _fetch_random_block too. Did you try it?

Reply:
Also, in case of exception, it will be handled as a value in `_fetch_latest_block` and lead to an error I guess.